libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -436,7 +476,14 @@ void AP_Baro::update(void)
             if (is_zero(ground_pressure) || isnan(ground_pressure) || isinf(ground_pressure)) {
                 sensors[i].ground_pressure = sensors[i].pressure;
             }
-            float altitude = get_altitude_difference(sensors[i].ground_pressure, sensors[i].pressure);
+            float altitude = sensors[i].altitude;
+            if(sensors[i].type == BARO_TYPE_AIR) {
+            	altitude = get_altitude_difference(sensors[i].ground_pressure, sensors[i].pressure);
+            } else if(sensors[i].type == BARO_TYPE_WATER) {
+            	//101325Pa is sea level air pressure, 9800 Pascal/ m depth in water.
+            	//No temperature or depth compensation for density of water.
+            	altitude = (sensors[i].ground_pressure - sensors[i].pressure) / 9800.0f / _specific_gravity;
```

Comment:
I've gone about it wrong the entire time. I've read the style guide, thanks for pointing it out to me. astyle has come in handy for fixing these things. I've also configured my editor to use spaces :).